homeassistant/components/blue_current/__init__.py

```diff
@@ -47,16 +44,15 @@ async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> b
     connector = Connector(hass, config_entry, client)
 
     try:
-        await connector.connect(api_token)
+        await client.validate_api_token(api_token)
     except InvalidApiToken as err:
         raise ConfigEntryAuthFailed("Invalid API token.") from err
     except BlueCurrentException as err:
         raise ConfigEntryNotReady from err
 
-    hass.async_create_background_task(connector.start_loop(), "blue_current-websocket")
-    await client.get_charge_points()
+    hass.async_create_background_task(connector.connect(), "blue_current-websocket")
```

Comment:
```suggestion
    config_entry.async_create_background_task(hass, connector.run_task(), "blue_current-websocket")
```

The following code can be removed as the background task is automatically cancelled on unload
```python
    config_entry.async_on_unload(connector.disconnect)
    async def _async_disconnect_websocket(_: Event) -> None:
        await connector.disconnect()
    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, _async_disconnect_websocket)
```